model.py

```diff
@@ -60,41 +83,52 @@ def __init__(self):
         # )
 
         self.vae_img_patch_embed = PatchEmbed(
-            height=LATENT_DIM,
-            width=LATENT_DIM,
-            patch_size=PATCH_DIM,
-            in_channels=LATENT_CHANNELS,
-            embed_dim=EMBEDDING_DIM,
+            height=self.latent_dim,
+            width=self.latent_dim,
+            patch_size=self.patch_dim,
+            in_channels=self.vae_latent_channels,
+            embed_dim=self.embed_dim,
             use_pos_embed=True,
         )
         
         self.timestep_embed = Timesteps(
-            EMBEDDING_DIM,
+            self.embed_dim,
             flip_sin_to_cos=True,
             downscale_freq_shift=0,
         )
-        
-        # TODO(2): Use official tokenizer and model / upgrade the tokenizer and model
-        # self.llama_tokenizer = LlamaTokenizerFast.from_pretrained(LLAMA_MODEL)
-        # self.llama = LlamaModel.from_pretrained(LLAMA_MODEL, device_map="auto", torch_dtype=torch.float16)
-        self.llama_tokenizer = AutoTokenizer.from_pretrained(TOKENIZER)
-        self.llama = AutoModel.from_pretrained(MODEL)
 
         self.llama_token_embs = self.llama.get_input_embeddings()  # TODO: May be a better way
 
         # LC*L*L/NP
-        self.transformer_img_proj = nn.Linear(EMBEDDING_DIM, LATENT_CHANNELS*LATENT_DIM*LATENT_DIM//NUM_PATCHES)
+        self.transformer_img_proj = nn.Linear(
+            self.embed_dim, self.vae_latent_channels * self.latent_dim * self.latent_dim // self.num_patches)
+    
+    def tokenize(self, text, device="cpu", padding="max_length", truncation=True, **tokenizer_kwargs):
+        tokenizer_output = self.llama_tokenizer(
+            text, padding=padding, truncation=truncation, max_length=self.context_size, **tokenizer_kwargs
+        )
+
+        # For now only return input_ids and attention_mask
+        input_ids = torch.tensor(tokenizer_output["input_ids"], device=device, dtype=torch.long)
+        attention_mask = torch.tensor(tokenizer_output["attention_mask"], device=device, dtype=torch.float)
+
+        return input_ids, attention_mask
```

Comment:
I know it's used as a mask of the attention, but I think calling it `pad_mask` here is more in-line with the output from a tokenizer. Feel free to override though.